main.tf

```
@@ -208,6 +210,26 @@ resource "google_container_cluster" "cluster" {
     }
   }
 
+
+  # --------------------------------------------------------------------------------------------------------------------
+  # AutoPilot node configuration
+  # --------------------------------------------------------------------------------------------------------------------
+
+
+  dynamic "node_pool_auto_config" {
+    for_each = var.node_pool_auto_config != null ? [1] : []
+
+    content {
+      dynamic "network_tags" {
+        for_each = var.node_pool_auto_config.network_tags.tags != null ? [1] : []
+
+        content {
+          tags = var.node_pool_auto_config.network_tags.tags
+        }
```

please see https://www.notion.so/Creating-new-Mineiros-Modules-e26d711cda7d43f39fe73d316eaf0d6e#d6f4820e6bd14c61bad85da5f3068cd3

```suggestion
    for_each = var.node_pool_auto_config != null ? [var.node_pool_auto_config] : []

    content {
      dynamic "network_tags" {
        for_each = try([node_pool_auto_config.value.network_tags], [])

        content {
          tags = try(network_tags.value.tags, null)
        }
```